src/Minsk/CodeAnalysis/Binding/BoundGlobalScope.cs

```diff
@@ -5,19 +5,17 @@ namespace Minsk.CodeAnalysis.Binding
 {
     internal sealed class BoundGlobalScope
     {
-        public BoundGlobalScope(BoundGlobalScope previous, ImmutableArray<Diagnostic> diagnostics, ImmutableArray<FunctionSymbol> functions, ImmutableArray<VariableSymbol> variables, ImmutableArray<BoundStatement> statements)
+        public BoundGlobalScope(BoundScope scope, ImmutableArray<(FunctionSymbol function, BoundBlockStatement body)> functionBodies, ImmutableArray<Diagnostic> diagnostics, ImmutableArray<BoundStatement> statements)
         {
-            Previous = previous;
+            Scope = scope;
+            FunctionBodies = functionBodies;
             Diagnostics = diagnostics;
-            Functions = functions;
-            Variables = variables;
             Statements = statements;
         }
 
-        public BoundGlobalScope Previous { get; }
+        public BoundScope Scope { get; }
+        public ImmutableArray<(FunctionSymbol function, BoundBlockStatement body)> FunctionBodies { get; }
         public ImmutableArray<Diagnostic> Diagnostics { get; }
-        public ImmutableArray<FunctionSymbol> Functions { get; }
-        public ImmutableArray<VariableSymbol> Variables { get; }
```

Comment:
I don't think this is useful anymore since it's accessible from the scope (and noone actually used these properties anyway after the refactoring I did).